### PR TITLE
✨Source Apify Dataset: Migrate Python CDK to Low-code CDK

### LINKED ISSUE #29116: Apify Dataset

### PR DESCRIPTION
**Pull request by btkcodedev**
## What
*Migrating Source Apify Dataset to Low-Code CDK*
Closes #29116

## How
*Developed using (Configuration Based Source) low-code CDK*

## Recommended reading order
1. `spec.yaml`
2. `manifest.yaml`
3. `schemas/*`

## 🚨 User Impact 🚨
- No breaking changes, migration to low-code
<details open><summary><strong>Migrate Connector: Source Apify Dataset</strong></summary>

## Tests
image

### Community member or Airbyter

- [x] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] Code reviews completed
- [x] Documentation updated
    - [x] Connector's `README.md`
    - [x] Connector's `bootstrap.md`
    - [x] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [x] `docs/integrations/README.md`
    - [x] `airbyte-integrations/builds.md`
- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

</details>

### CODE REVIEW

**Comment by btkcodedev** [comment]
Config
{
  "token": "apify_api_8UPAaECr6WGLasifLAicj0tDziE64E4j40dT",
  "start_date": "2023-08-25T00:00:00.244Z"
}

**Comment by btkcodedev** [comment]
Checks: 
Added Paginator ✅ 
Added Incremental Sync ✅ 
Updated Version in metadata, Dockerfile and docs ✅

File: docs/integrations/sources/apify-dataset.md [hunk 1]
```diff
@@ -37,16 +37,18 @@ The Apify dataset connector uses [Apify Python Client](https://docs.apify.com/ap
...
-| 0.1.11 | 2022-04-27 | [12397](https://github.com/airbytehq/airbyte/pull/12397) | No changes. Used connector to test publish workflow changes. |
-| 0.1.9 | 2022-04-05 | [PR\#11712](https://github.com/airbytehq/airbyte/pull/11712) | No changes from 0.1.4. Used connector to test publish workflow changes. |
-| 0.1.4 | 2021-12-23 | [PR\#8434](https://github.com/airbytehq/airbyte/pull/8434) | Update fields in source-connectors specifications |
-| 0.1.2 | 2021-11-08 | [PR\#7499](https://github.com/airbytehq/airbyte/pull/7499) | Remove base-python dependencies |
-| 0.1.0 | 2021-07-29 | [PR\#5069](https://github.com/airbytehq/airbyte/pull/5069) | Initial version of the connector |
+| Version   | Date        | Pull Request                                                  | Subject                                                                     |
+| :-------- | :---------- | :------------------------------------------------------------ | :-------------------------------------------------------------------------- |
+|   0.3.0   | 2023-08-25  | [TBD](https://github.com/airbytehq/airbyte/pull/TBD)          | Migrate to lowcode                                                          |
```
Comment (marcosmarxm) [inline docs/integrations/sources/apify-dataset.md:47]: Update this.

File: airbyte-integrations/connectors/source-apify-dataset/source_apify_dataset/manifest.yaml [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    cursor_field: "{{ parameters.incremental_cursor }}"
+    datetime_format: "%Y-%m-%dT%H:%M:%S.%f%z"
+    cursor_granularity: "PT0.000001S"
+    start_datetime:
+      datetime: "{{ config['start_date'] }}"
+      datetime_format: "%Y-%m-%dT%H:%M:%S.%f%z"
+    end_datetime:
+      datetime: "{{ today_utc() }}"
+      datetime_format: "%Y-%m-%d"
+    step: "P1M"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-apify-dataset/source_apify_dataset/manifest.yaml:57]: Can you reference Apify documentation where you can use incremental?

File: airbyte-integrations/connectors/source-apify-dataset/source_apify_dataset/manifest.yaml [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+  retriever:
+    type: SimpleRetriever
+    record_selector:
+      $ref: "#/definitions/selector"
+    paginator:
+      type: "NoPagination"
+    requester:
+      $ref: "#/definitions/requester"
+
+  base_paginator:
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-apify-dataset/source_apify_dataset/manifest.yaml:27]: Not used anywhere. Please make sure to use the paginator.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Left some comments.

**Comment by btkcodedev** [comment]
Reolved comments, somehow my apifydataset account got disabled. Thus couldn't test the changes :(

**Review by marcosmarxm — APPROVED** [review]
Thanks @btkcodedev

**Pull request merged**